test/Kestrel.InMemory.FunctionalTests/Http2/Http2TimeoutTests.cs

```diff
@@ -208,40 +209,50 @@ public async Task AbortedStream_ResetsAndDrainsRequest_RefusesFramesAfterCooldow
 
             await WaitForStreamErrorAsync(1, Http2ErrorCode.INTERNAL_ERROR, "The connection was aborted by the application.");
 
-            // There's a race when the appfunc is exiting about how soon it unregisters the stream.
-            for (var i = 0; i < 10; i++)
+            var cts = new CancellationTokenSource();
+
+            async Task AdvanceClockAndSendFrames()
             {
-                await SendDataAsync(1, new byte[100], endStream: false);
+                // There's a race when the appfunc is exiting about how soon it unregisters the stream, so retry until success.
+                while (!cts.Token.IsCancellationRequested)
+                {
+                    // Just past the timeout
+                    mockSystemClock.UtcNow += Constants.RequestBodyDrainTimeout + TimeSpan.FromTicks(1);
+                    (_connection as IRequestProcessor).Tick(mockSystemClock.UtcNow);
+
+                    // Send an extra frame to make it fail
+                    switch (finalFrameType)
+                    {
+                        case Http2FrameType.DATA:
+                            await SendDataAsync(1, new byte[100], endStream: true);
+                            break;
+
+                        case Http2FrameType.HEADERS:
+                            await SendHeadersAsync(1, Http2HeadersFrameFlags.END_STREAM | Http2HeadersFrameFlags.END_HEADERS, _requestTrailers);
```

Comment:
Same problem here, this shouldn't be retried as is.

Reply:
Good catches. I meant to change this and forgot.